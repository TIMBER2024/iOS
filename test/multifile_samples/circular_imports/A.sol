import "./B.sol";

/// #invariant true;
contract A {
  function a(B b) external returns (B) { return b; }
}
