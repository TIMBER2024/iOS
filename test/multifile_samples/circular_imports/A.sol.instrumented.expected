/// This file is auto-generated by Scribble and shouldn't be edited directly.
/// Use --disarm prior to make any changes.
import "./B.sol";

/// #invariant true;
contract A {
    function a(B b) external returns (B RET_0) {
        __ScribbleUtilsLib__16.setInContract(true);
        RET_0 = _original_A_a(b);
        __scribble_check_state_invariants();
        __ScribbleUtilsLib__16.setInContract(false);
    }

    function _original_A_a(B b) internal returns (B) {
        return b;
    }

    /// Check only the current contract's state invariants
    function __scribble_A_check_state_invariants_internal() internal {
        unchecked {
            if (!(true)) {
                emit __ScribbleUtilsLib__16.AssertionFailed("000688:0066:000 0: ");
                assert(false);
            }
        }
    }

    /// Check the state invariant for the current contract and all its bases
    function __scribble_check_state_invariants() virtual internal {
        __scribble_A_check_state_invariants_internal();
    }

    constructor() {
        __ScribbleUtilsLib__16.setInContract(true);
        __scribble_check_state_invariants();
        __ScribbleUtilsLib__16.setInContract(false);
    }
}

library __ScribbleUtilsLib__16 {
    event AssertionFailed(string message);

    event AssertionFailedData(int eventId, bytes encodingData);

    function assertionFailed(string memory arg_0) internal {
        emit AssertionFailed(arg_0);
    }

    function assertionFailedData(int arg_0, bytes memory arg_1) internal {
        emit AssertionFailedData(arg_0, arg_1);
    }

    function isInContract() internal returns (bool res) {
        assembly {
            res := sload(0x5f0b92cf9616afdee4f4136f66393f1343b027f01be893fa569eb2e2b667a40c)
        }
    }

    function setInContract(bool v) internal {
        assembly {
            sstore(0x5f0b92cf9616afdee4f4136f66393f1343b027f01be893fa569eb2e2b667a40c, v)
        }
    }
}